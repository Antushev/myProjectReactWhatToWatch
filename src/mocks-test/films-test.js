import {generateFilm, generateFilms} from '../mocks/films.js';

const FILMS_NUMBER = 10;

const film = generateFilm(1);
const films = generateFilms(FILMS_NUMBER);

export {film, films};
