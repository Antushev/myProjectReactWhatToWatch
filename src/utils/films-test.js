const films = [
  {
    id: 1,
    name: `Хоббит: Пустошь Смауга`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2001
  },
  {
    id: 2,
    name: `Интерстеллар`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2000
  },
  {
    id: 3,
    name: `Спасти рядоого Райана`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2004
  },
  {
    id: 4,
    name: `Хоббит: Пустошь Смауга`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2014
  },
  {
    id: 5,
    name: `Власталин колец: Возвращение короля`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2014
  },
  {
    id: 6,
    name: `Хоббит: Пустошь Смауга`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2019
  },
  {
    id: 7,
    name: `Хоббит: Нежданное путешествие`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2020
  },
  {
    id: 8,
    name: `Хоббит: Пустошь Смауга`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2018
  }
];

const film = {
  id: 1,
  name: `Интерстеллар`,
  picture: `img/pictrue.jpg`,
  genre: `Adventure`,
  date: 2014
};

export {film, films};
