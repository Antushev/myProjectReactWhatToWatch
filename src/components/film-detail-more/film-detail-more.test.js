import React from 'react';
import renderer from 'react-test-renderer';

import {film} from './../../mocks-test/films-test.js';

import FilmDetailMore from './film-detail-more.jsx';

describe(`FilmDetailMoreComponent`, () => {
  it(`FilmDetailMoreComponentSnapshot`, () => {
    const tree = renderer
      .create(
          <FilmDetailMore
            film={film}
            renderTabs={() => {}}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
