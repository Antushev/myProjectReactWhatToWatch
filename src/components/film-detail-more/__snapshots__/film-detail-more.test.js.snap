// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDetailMoreComponent FilmDetailMoreComponentSnapshot 1`] = `
<div
  className="movie-card__wrap movie-card__translate-top"
>
  <div
    className="movie-card__info"
  >
    <div
      className="movie-card__poster movie-card__poster--big"
    >
      <img
        alt="Интерстеллар"
        height="327"
        src="img/aviator.jpg"
        width="218"
      />
    </div>
    <div
      className="movie-card__desc"
    >
      <div
        className="movie-card__text movie-card__row"
      >
        <div
          className="movie-card__text-col"
        >
          <p
            className="movie-card__details-item"
          >
            <strong
              className="movie-card__details-name"
            >
              Director
            </strong>
            <span
              className="movie-card__details-value"
            >
              Джон
            </span>
          </p>
          <p
            className="movie-card__details-item"
          >
            <strong
              className="movie-card__details-name"
            >
              Starring
            </strong>
            <span
              className="movie-card__details-value"
            >
              Гоша, 
Щас Зе Енд, 
Куллер, 
Райан Оркестр
            </span>
          </p>
        </div>
        <div
          className="movie-card__text-col"
        >
          <p
            className="movie-card__details-item"
          >
            <strong
              className="movie-card__details-name"
            >
              Run Time
            </strong>
            <span
              className="movie-card__details-value"
            >
              1h 40m
            </span>
          </p>
          <p
            className="movie-card__details-item"
          >
            <strong
              className="movie-card__details-name"
            >
              Genre
            </strong>
            <span
              className="movie-card__details-value"
            >
              Adventure
            </span>
          </p>
          <p
            className="movie-card__details-item"
          >
            <strong
              className="movie-card__details-name"
            >
              Released
            </strong>
            <span
              className="movie-card__details-value"
            >
              2001
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
